sklearn/neighbors/base.py

```diff
@@ -421,6 +454,174 @@ class from an array representing our data set and ask who's
                 return dist, neigh_ind
             return neigh_ind
 
+    def masked_kneighbors(self, X=None, n_neighbors=None, return_distance=True,
```

Comment:
Is the main point in duplicating this to pass `force_all_finite=False`? I don't think it's necessary.

Rather, I think we should be leaving the data validation to the `pairwise_distances` and ball/binary tree (which seem to validate when queried, but not when constructed), and remove it from here / make it minimal. I admit that getting this right might be tricky, but I am not happy with a solution that duplicates `kneighbors` (and why not `radius_neighbors` also?) for the sake of validating more leniently.